### PR TITLE
Adviser typeahead filter for collection pages

### PR DESCRIPTION
**Pull request by ian-leggett**
## Description of change
This is the adviser typeahead filters component that will be used with the new collection pages #2905 Its very similar to our previous adviser typeahead component, the difference is that it's not coupled to the context API, it uses the Task component and falls inline with our [React/redux architecture](https://github.com/uktrade/data-hub-frontend/blob/master/docs/Redux%20and%20Saga.md).

## Test instructions
Run Storybook and you should be able to see this component under "filters"

## Screenshots
Screenshot 2020-10-05 at 11 40 07

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        placeholder="Search advisers..."
+        noOptionsMessage={() => <span>No advisers found</span>}
+        taskProps={{
+          name: 'task name 2',
+          id: 'id2',
+        }}
+      />
+    </div>
+  ))
+  .add('Pre-selected option', () => (
```
Comment (rachaelcodes) [inline src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx:91]: Should we be able to see an option pre-selected in the Story? It's not showing up for me (here or in the other one)

**Comment by rachaelcodes** [comment]
There's a possible bug - on the 'single select with pre-selected option', I can't see anything selected:
Screenshot 2020-10-13 at 14 24 11

File: src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      <SearchMesssage />
+      <FilterAdvisersTypeaheadWithState
+        label="Advisers"
+        isMulti={true}
+        closeMenuOnSelect={false}
+        name="example-3"
+        loadOptions={loadOptions}
+        placeholder="Search advisers..."
+        noOptionsMessage={() => <span>No advisers found</span>}
+        selectedAdvisers={asyncOptions[2]}
```
Comment (mforner13) [inline src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx:102]: Should you be able to remove the pre-selected option? As can't currently do this in the story.

**Review by mforner13 — COMMENTED** [review]
Looks good! A couple of comments, and one thing which I've noticed from testing it out in storybook:

Sometimes when I begin typing in the field, the beginning of the text moves across slightly and some of it is hidden. I can't re-create it again now though, so am wondering whether it was just something weird on my machine? 🤔  😛 

image

**Comment by mforner13** [comment]
> There's a possible bug - on the 'single select with pre-selected option', I can't see anything selected:
> Screenshot 2020-10-13 at 14 24 11

@rachaelcodes Was that looking at the `Typeahead` component instead of `Filters/Adviser Typeahead` ?

File: src/client/components/Filters/elements/FilterAdvisersTypeahead/usage.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+FilterAdvisersTypeahead
+=========
+
+### Description
+
+This is a typeahead is to be used in the collection list pages, this component depends on the [Task component](https://github.com/uktrade/data-hub-frontend/tree/master/src/client/components/Task) and should be used in conjuction with our [React/redux architecture](https://github.com/uktrade/data-hub-frontend/blob/master/docs/Redux%20and%20Saga.md).
```
[the remark this answers is not in the document]
Comment (ian-leggett) [reply src/client/components/Filters/elements/FilterAdvisersTypeahead/usage.md:6]: Removed this whole file as after speaking to Mark we feel that these can get out of date fast and don't really help us.

File: src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx (hunk 2 above)
Comment (ian-leggett) [reply src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx:102]: It does actually work if used in Data Hub but I agree this should be reflected in Storybook 👍

**Comment by ian-leggett** [comment]
> > There's a possible bug - on the 'single select with pre-selected option', I can't see anything selected:
> > Screenshot 2020-10-13 at 14 24 11
> 
> @rachaelcodes Was that looking at the `Typeahead` component instead of `Filters/Adviser Typeahead` ?
> 
> *Ahh unless the changes to `Typeahead/styles` has broken Typeahead single select!

This doesn't seem to be working even on the live version, I think this is a bug outside this PR.

**Comment by ian-leggett** [comment]
> Looks good! A couple of comments, and one thing which I've noticed from testing it out in storybook:
> 
> Sometimes when I begin typing in the field, the beginning of the text moves across slightly and some of it is hidden. I can't re-create it again now though, so am wondering whether it was just something weird on my machine? 🤔 😛
> 
> image

This is strange as I have not seen this before and cannot create the bug.

**Comment by ian-leggett** [comment]
> There's a possible bug - on the 'single select with pre-selected option', I can't see anything selected:
> Screenshot 2020-10-13 at 14 24 11

This was broken in the live version, all fixed now 👍

**Comment by rachaelcodes** [comment]
We're getting an accessibility issue raised in both Storybook and axe testing about the label not being linked to the input: 
Screenshot 2020-10-20 at 11 56 21

**Review by mforner13 — APPROVED** [review]
Looks good! 😄

File: src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
+/* eslint "react/destructuring-assignment": 0, react/prop-types: 0 */
+
+import React, { useState } from 'react'
+import { storiesOf } from '@storybook/react'
+
+import FilterAdvisersTypeahead from '../elements/FilterAdvisersTypeahead'
+// import exampleReadme from '../elements/FilterAdvisersTypeahead/example.md'
```
Comment (rachaelcodes) [inline src/client/components/Filters/__stories__/FilterAdvisersTypeahead.stories.jsx:7]: Do we need this comment?

**Review by rachaelcodes — APPROVED** [review]

**Pull request merged**